### PR TITLE
Feature/registration

### PR DESCRIPTION
**Pull request by forever-Agriculture**
### registration
*1. simple registration added*
*2. simple login added*
*3. simple validation and error notifications added*
*4. general hashing system for flask and django implemented*

### CODE REVIEW

File: backend/models/users.py [hunk 1]
```diff
@@ -4,7 +4,8 @@
 from sqlalchemy.ext.hybrid import hybrid_property
 from sqlalchemy.sql.functions import func
 
-from backend.app import bcrypt, db
+from backend.app import db
```
Comment (NewIvanIvanov) [inline backend/models/users.py]: wrong order of import

File: client/apps/city_issues/models/users.py [hunk 2]
```diff
@@ -73,16 +79,20 @@ def password(self):
         return self.hashed_password
 
     @password.setter
-    def set_hashed_password(self, raw_password):
-        super(User, self).set_password(raw_password)
+    def password(self, raw_password):
+        self.set_password(raw_password)
```
Comment (NewIvanIvanov) [inline client/apps/city_issues/models/users.py:82]: why we use this function ? its only call "set_password()" ?
why we cant use just "set_password()"

File: client/apps/city_issues/views.py [hunk 3]
```diff
@@ -3,13 +3,13 @@
 """
 # -*- coding: utf-8 -*-
 from django.views.generic import CreateView
+from django.views.generic.base import TemplateView
 from django.core import serializers
+from django.contrib import messages
```
Comment (NewIvanIvanov) [inline client/apps/city_issues/views.py]: wrong import order

File: client/apps/city_issues/user_managers.py [hunk 4]
```diff
@@ -18,7 +18,7 @@ def _create_user(self, name, email, password, **extra_fields):
         user.save(using=self._db)
```
Comment (NewIvanIvanov) [inline client/apps/city_issues/user_managers.py:18]: why we use "using=self._db". we use only one database currently

File: requirements.txt [hunk 5]
```diff
@@ -1,32 +1,45 @@
 alembic==0.9.6
+astroid==1.5.3
```
Comment (NewIvanIvanov) [inline requirements.txt]: pleace clean requirements.txt of pylint packages

**Review by NewIvanIvanov — CHANGES_REQUESTED** [review]

File: backend/models/users.py (hunk 1 above)
Comment (forever-Agriculture) [reply backend/models/users.py]: +

File: client/apps/city_issues/models/users.py (hunk 2 above)
Comment (forever-Agriculture) [reply client/apps/city_issues/models/users.py:82]: impossible, property requires a setter and a setter with the same name, if i got ur point

File: client/apps/city_issues/user_managers.py (hunk 4 above)
Comment (forever-Agriculture) [reply client/apps/city_issues/user_managers.py:18]: yes, we use just 1 database and simply saving data doesn't create a new database

Comment (forever-Agriculture) [reply client/apps/city_issues/user_managers.py:18]: it's just a custom user saving method, more accurately for correct overriding of django default user manager

File: client/apps/city_issues/views.py (hunk 3 above)
Comment (forever-Agriculture) [reply client/apps/city_issues/views.py]: done

File: requirements.txt (hunk 5 above)
Comment (forever-Agriculture) [reply requirements.txt]: done

File: client/apps/city_issues/user_managers.py (hunk 4 above)
Comment (forever-Agriculture) [reply client/apps/city_issues/user_managers.py:18]: https://github.com/django/django/blob/e283c1a2675d1336aba0e8141ed94efa3ac68b04/django/contrib/auth/models.py#L144

File: client/apps/city_issues/models/users.py (hunk 2 above)
Comment (forever-Agriculture) [reply client/apps/city_issues/models/users.py:82]: https://docs.djangoproject.com/en/1.11/topics/auth/customizing/#specifying-a-custom-user-model
https://docs.python.org/2/howto/descriptor.html#properties

**Review by NewIvanIvanov — APPROVED** [review]

**Pull request merged**